test/extended/networking/network_segmentation.go

```diff
@@ -1890,6 +1890,12 @@ type networkAttachmentConfigParams struct {
 	role               string
 }
 
+// workloadNetworkConfig contains workload-specific network customizations
+type workloadNetworkConfig struct {
```

Comment:
Maybe it makes sense to "prepare" for the future and assume we want to one day have multiple of these. For that, you'd have to add here the workload name. If we ever want to extend this to pods (god forbid) we might actually have to add the resource type as well. Anyway, I'm digressing ... 

We can postpone it until it is actually needed.